RavenDB-22346 Remove the take parameter for `SortingMultiMatch`

maciejaszyk opened the pull request:
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-22346 

### Additional description

Remove the take parameter for `SortingMultiMatch` since we've to do a sort on the whole set using all comparers to get the result.

### Type of change

- [x] Bug fix
- [ ] Regression bug fix
- [ ] Optimization
- [ ] New feature

### How risky is the change?

- [ ] Low 
- [ ] Moderate 
- [ ] High
- [ ] Not relevant

### Backward compatibility

- [x] Non breaking change
- [ ] Ensured. Please explain how has it been implemented?
- [ ] Breaking change
- [ ] Not relevant

### Is it platform specific issue?

- [ ] Yes. Please list the affected platforms.
- [x] No

### Documentation update

- [ ] This change requires a documentation update. Please mark the issue on YouTrack using `Documentation Required` tag.
- [x] No documentation update is needed 

### Testing by Contributor

- [ ] Tests have been added that prove the fix is effective or that the feature works
- [ ] Internal classes added to the test class (e.g. entity or index definition classes) have the lowest possible access modifier (preferable `private`) 
- [ ] It has been verified by manual testing

### Testing by RavenDB QA team

- [ ] This change requires a special QA testing due to possible performance or resources usage implications (CPU, memory, IO). Please mark the issue on YouTrack using `QA Required` tag.
- [x] No special testing by RavenDB QA team is needed

### Is there any existing behavior change of other features due to this change?

- [ ] Yes. Please list the affected features/subsystems and provide appropriate explanation
- [x] No

### UI work

- [ ] It requires further work in the Studio. Please mark the issue on YouTrack using `Studio Required` tag.
- [x] No UI work is needed

maciejaszyk committed:
RavenDB-22346 Remove the take parameter for `SortingMultiMatch` since we've to do a sort on the whole set using all comparers to get the result.

arekpalinski commented:
Please check https://jenkins.hibernatingrhinos.com/job/PR_Tests/10653/testReport/junit/SlowTests.Server.Documents.Queries.Dynamic.Map/DynamicQueriesEnumsNestedFieldsAndCollections/Query_on_collection_options___DatabaseMode___Sharded___SearchEngineMode___Corax_/

maciejaszyk committed:
RavenDB-22346 Tests adjustment: Query without order by in sharding mode does not guarantee order.

maciejaszyk commented:
> Please check https://jenkins.hibernatingrhinos.com/job/PR_Tests/10653/testReport/junit/SlowTests.Server.Documents.Queries.Dynamic.Map/DynamicQueriesEnumsNestedFieldsAndCollections/Query_on_collection_options___DatabaseMode___Sharded___SearchEngineMode___Corax_/

Fixed, adjusted the tests for sharding mode.
